utils/s3/aws_credentials_provider.hh

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+#include "utils/s3/creds.hh"
+
+namespace aws {
+/*
+ * Abstract class for retrieving AWS credentials. Create a derived class from this to allow various methods of storing and retrieving credentials. Examples
+ * would be cognito-identity, some encrypted store etc...
+ */
+class aws_credentials_provider {
```

Comment:
This class should probably hold an instance of `s3::endpoint_config::aws_config`, return it in `get_aws_credentials()` and only virtually `reload()` it if `is_time_to_refresh()`.

for example:
```cpp
get_aws_credentials() {
  if (is_time_to_refresh()) {
     _aws_config  = reload();
  }
  return _aws_config;
}
```

But other mechanics are ok too :-).